modules/aerodyn/src/FVW_VortexTools.f90

```diff
@@ -51,6 +52,127 @@ module FVW_VortexTools
+   !! Estimtes the regularization parameters based on the length of the tip region..
+   subroutine PlaceTipRoot(n, Gamma_b, xV, Eps, iRoot, iTip, Gamma_max, EpsTip, EpsRoot)
+      integer(IntKi),               intent(in)  :: n           !< number of vortex points
+      real(ReKi), dimension(n)    , intent(in)  :: Gamma_b     !< Wake panel circulation
+      real(Reki), dimension(3,n+1), intent(in)  :: xV          !< Vortex point nodal coordinates
+      real(Reki), dimension(3,n  ), intent(in)  :: Eps         !< Vortex panels epsilon
+      integer(IntKi),               intent(inout) :: iRoot, iTip !< Index of tip and root vortex, if <0, they are computed
+      real(ReKi),                   intent(out) :: Gamma_max
+      real(Reki),                   intent(out) :: EpsTip      !< Regularization of tip
+      real(Reki),                   intent(out) :: EpsRoot     !< Regularization of root
+      real(ReKi), dimension(n+1) :: Gamma_t
+      real(ReKi), dimension(n+1) :: s
+      real(ReKi) :: rRoot, rTip, lTip, lRoot
+      integer(IntKi) :: i10, i90, iTipPanel,iRootPanel
+
+      if (n>2) then
+         Gamma_max = maxval(Gamma_b,1)
+         call CurvilinearDist(n+1, xV, s)
+         if (iTip<0) then
+
+            ! Find position of tip and root
+            call GammaTrailed(n, Gamma_b, Gamma_t)
+
+            ! If circulation is constant then use first and last
+            if(sum(abs(Gamma_t(:)))/(n+1)<1e-6) then
+               iTip =n+1
+               iRoot=1
+            else
+               i10 = minloc(abs(s(:)-0.15*s(n+1)),1)
+               i90 = minloc(abs(s(:)-0.85*s(n+1)),1)
+
+               rTip  = sum(Gamma_t(i90:)  * (s(i90:)))/ sum(Gamma_t(i90:))
+               rRoot = sum(Gamma_t(1:i10) * s(1:i10)) / sum(Gamma_t(1:i10))
+               iTip  = minloc(abs(rTip - s), 1) ! NOTE:  not accurate since epsilon has one dimension less..
+               iRoot = minloc(abs(rRoot - s), 1)
+               iTip  = max(min(iTip,n+1), i90)
+               iRoot = min(max(iRoot,1) , i10)
+            endif
+         endif
+         rTip  = s(iTip)
+         rRoot = s(iRoot)
+         iTipPanel  = max(min(iTip,n), 1)
+         iRootPanel = min(max(iRoot,1),n)
+         ! Mean regularization at the tip and root
+         EpsTip  = sum(Eps(1,iTipPanel:))  /(n-iTipPanel+1)
+         EpsRoot = sum(Eps(1,1:iRootPanel))/(iRootPanel)
+         ! Scaling based on the "length" of the vortex, this will need further tuning
+         lTip  = (s(n+1)-rTip )/3.14 ! Approximate radius if the tip has done half a turn
+         lRoot = (rRoot       )/3.14
+         EpsTip  = 1.3*(lTip+EpsTip) ! Tuning factors
+         EpsRoot = 1.7*(lRoot+EpsRoot)
+      else
+         iTip =n+1
+         iRoot=1
+         EpsTip   = Eps(1,iTip-1)
+         EpsRoot  = Eps(1,iRoot)
+      endif
+   endsubroutine PlaceTipRoot
+
+
+   !> Flatten/ravel a 3D grid of vectors (each of size n)
+   subroutine FlattenValues(GridValues, FlatValues, iHeadP)
```

Comment:
Wouldn't it be possible to use the Fortran intrinsic `reshape` to do what `FlattenValues` and `DeflateValues` do?

```
FlatValues(:,:) = reshape( GridValues, (/ size(gridvalues,1), size(gridvalues,4)*size(gridvalues,3)*size(gridvalues,2) /) )
```
(or something along those lines -- I don't remember the exact syntax)

Reply:
From my teaching and experience I've avoided using intrinsic functions for large datasets, some of them are using the stack and fail without reporting errors. I've been told these simple loops are easy for the compiler to optimize. But 
in that case you might be right, and I don't know which one would perform best!!! This function has a little "bonus" that allows me to flatten several grids into one array by calling this function several times for different arrays, changing iHeadP. I'm using this to pack all the control points from all the user-grids into one array that is then sent to the induced velocity function.